keras/backend/tensorflow_backend.py

```diff
@@ -570,6 +570,26 @@ def categorical_crossentropy(output, target, from_logits=False):
         return tf.nn.softmax_cross_entropy_with_logits(output, target)
 
 
+def categorical_crossentropy_one_hot(output, target, from_logits=False):
+    '''Note: tf.nn.sparse_softmax_cross_entropy_with_logits
+    expects logits, Keras expects prbabilities.
+    '''
+    if not from_logits:
+        output = tf.clip_by_value(output, tf.cast(_EPSILON, dtype=_FLOATX),
```

Comment:
I had previously just done `output = tf.log(output)`, and it had worked, but I wasn't very clear on the math of moving between logits and probabilities so I decided to just copy the conversion from your implementation of `binary_crossentropy`.